Files/Filesystem/ListedItem.cs

```diff
@@ -160,21 +160,73 @@ public bool IsImage()
             return false;
         }
 
-        public ListedItem(string folderRelativeId)
+        /// <summary>
+        /// Create an item object, optionally with an explicitly-specified dateReturnFormat.
+        /// </summary>
+        /// <param name="folderRelativeId"></param>
+        /// <param name="dateReturnFormat">Specify a date return format to reduce redundant checks of this setting.</param>
+        public ListedItem(string folderRelativeId, string dateReturnFormat = null)
         {
             FolderRelativeId = folderRelativeId;
+            if (dateReturnFormat != null)
+            {
+                DateReturnFormat = dateReturnFormat;
+            }
+            else
+            {
+                ApplicationDataContainer localSettings = ApplicationData.Current.LocalSettings;
+                string returnformat = Enum.Parse<TimeStyle>(localSettings.Values[LocalSettings.DateTimeFormat].ToString()) == TimeStyle.Application ? "D" : "g";
+                DateReturnFormat = returnformat;
+            }
         }
 
-        public static string GetFriendlyDate(DateTimeOffset d)
+        private string DateReturnFormat { get; }
+
+        private string GetFriendlyDate(DateTimeOffset d)
         {
-            ApplicationDataContainer localSettings = ApplicationData.Current.LocalSettings;
             var elapsed = DateTimeOffset.Now - d;
 
-            string returnformat = Enum.Parse<TimeStyle>(localSettings.Values[LocalSettings.DateTimeFormat].ToString()) == TimeStyle.Application ? "D" : "g";
+            if (elapsed.TotalDays > 7)
```

Comment:
Can you use a switch statement here?

Reply:
This is the way the code already was. And switch statements lack support for conditionals